# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmFrameIter.h
# Commit: 62480de389ff
# Full Hash: 62480de389ffba353998de24760993b2e1531232
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-11 15:10:42
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth. CLOSED TREE
#   
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
# ==============================================================================

diff -r f2e486f1be17 -r 62480de389ff js/src/wasm/WasmFrameIter.h
--- a/js/src/wasm/WasmFrameIter.h	Thu Aug 27 10:30:46 2020 +0000
+++ b/js/src/wasm/WasmFrameIter.h	Thu Aug 27 10:53:04 2020 +0000
@@ -65,7 +65,7 @@
   const CodeRange* codeRange_;
   unsigned lineOrBytecode_;
   Frame* fp_;
-  TlsData* tls_;
+  const TlsData* tls_;
   uint8_t* unwoundIonCallerFP_;
   jit::FrameType unwoundIonFrameType_;
   Unwind unwind_;
@@ -96,7 +96,7 @@
   jit::FrameType unwoundIonFrameType() const;
   uint8_t* unwoundIonCallerFP() const { return unwoundIonCallerFP_; }
   Frame* frame() const { return fp_; }
-  TlsData* tls() const { return tls_; }
+  const TlsData* tls() const { return tls_; }
 
   // Returns the address of the next instruction that will execute in this
   // frame, once control returns to this frame.
@@ -237,6 +237,7 @@
 
 // Iterates through frames for either possible cross-instance call or an entry
 // stub to obtain tls that corresponds to the passed fp.
+const TlsData* GetNearestEffectiveTls(const Frame* fp);
 TlsData* GetNearestEffectiveTls(Frame* fp);
 
 // Describes register state and associated code at a given call frame.